File: Project/ShoppingCart/ShoppingCart/Models/Product.cs
using ShoppingCart.Enum;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ShoppingCart.Models
{
    public class Product
    {
        #region Properties
        public int Id { get; set; }

        [ForeignKey("Category")]
        public int CategoryId { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [Required]
        [StringLength(250)]
        public string Description { get; set; }

        [Required]
        public double Discount { get; set; }

        [Required]
        public double Price { get; set; }

        [StringLength(100)]
        public string DefaultImage { get; set; }

        [NotMapped]
        public string Size { get; set; }
        
        [NotMapped]
        public int BaseProduct { get; set; }
        #endregion
    }
}

File: Project/ShoppingCartClient/src/app/Models/Product.ts
export class Product{
    id: number;
    categoryId: number;
    name: string;
    description: string;
    discount: number;
    price: number;
    defaultImage: string;
    size: string;
    qty: number
    baseProduct: number;


    constructor(id: number, categoryId: number, name: string, description: string, 
        discount: number,  price: number,  defaultImage: string,  size: string, qty: number = 1, baseProduct: number){
            this.id = id;
            this.categoryId = categoryId;
            this.name = name;
            this.description = description;
            this.discount = discount;
            this.price = price;
            this.defaultImage = defaultImage;
            this.size = size;
            this.qty = qty;
    }
}